classes/filter.php

```diff
@@ -232,6 +232,23 @@ public function save() {
         }
     }
 
+    public function save_default() {
```

Comment:
* Could you add the PHP Documentation
* Do you think we need to duplicate so much of the code from `save`? All of it is identical, except for the rules.
* How about abusing the `courseid` field, where 0 would mean default? So we do not need a new table.
* Alternatively, we could keep the separate table if we think more options could be added to the default rules at a later stage, the first thing that comes to mind is the `editable` flag, which I'd like to keep (see more below).

Yet, another option is to extend the `filter` class for a `filter_default` class.